### ISSUE TITLE
Taildrop: 502 Bad Gateway attempting to send a file from a Linux server to Android

### ISSUE THREAD

**Issue by GOATS2K** [tailscale/tailscale#4293]
### What is the issue?

`tailscale file cp --verbose loner.mp4 pixel-6-pro:`

```
2022/03/27 01:49:06 sending "loner.mp4" to pixel-6-pro/100.110.71.64/ngH2ag7CNTRL ...
502 Bad Gateway:
```

I expected the file to be sent.

### Steps to reproduce

I ran the `tailscaile file cp` command on my server, running Ubuntu 20.04.

Attempted sending the file to a Pixel 6 Pro, running Android 12.

### OS

Linux, Android

### OS version

Ubuntu 20.04

### Tailscale version

1.22.2

### Bug report

BUG-a553851f8705625b010d90125777946fb62e76eb1e8a768368bed8b1b8e1dc1d-20220327020308Z-ab2e23156be1b736

**Comment by bradfitz** [comment]
[Notes to self]

on the sending side:
```
2022-03-27 01:49:07.021133382 +0000 UTC: http: proxy error: dial tcp 100.110.71.64:42858: connect: connection refused
```

on the receiving side (c809a371970690b9dd6740b0b6caca05bf07896398c8747e102d1e513dbd406f) a bit earlier, perhaps from other attempt:
```
2022-03-27 01:18:16.815868906 +0000 UTC: 12.8M/236.5M peerapi.Accept: accept tcp6 [fd7a:115c:a1e0:ab12:4843:cd96:626e:4740]:42858: accept: invalid argument
2022-03-27 01:18:16.824863331 +0000 UTC: 12.8M/236.5M peerapi.Accept: accept tcp4 100.110.71.64:42858: accept: invalid argument
```

In bd90781b34d2558f27196194ba871f1cda60ffe6 we started using netstack for peerapi, though, for everywhere but iOS, so that should include Android.

**Comment by bradfitz** [comment]
@GOATS2K, if this is reproducible, could you tell me what this says from Linux...

First, `tailscale ping -tsmp 100.110.71.64` and you'll see something like:

```
pong from pixel-6-pro (100.110.71.64, 42858) via TSMP in 100ms
```

That number after the comma is the port number (which might change over time) of the "peerapi" service that Taildrop uses.

Then try to curl that port number...

```
$ curl -v http://100.110.71.64:42858
```

What does that say?

**Comment by GOATS2K** [comment]
Oddly enough, file transfers work now!

```
*   Trying 100.110.71.64:42858...
* TCP_NODELAY set
* Connected to 100.110.71.64 (100.110.71.64) port 42858 (#0)
> GET / HTTP/1.1
> Host: 100.110.71.64:42858
> User-Agent: curl/7.68.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Date: Sun, 27 Mar 2022 18:53:21 GMT
< Content-Length: 225
< Content-Type: text/html; charset=utf-8
<
<html>
<meta name="viewport" content="width=device-width, initial-scale=1">
<body>
<h1>Hello, John Patrick Glattetre (100.90.87.41)</h1>
This is my Tailscale device. Your device is neptune.
<p>You are the owner of this node.
* Connection #0 to host 100.110.71.64 left intact
```

**Comment by bradfitz** [comment]
This came up elsewhere, so copying a comment I wrote:

> my last hunch was that on Android we fail to listen on the peerapi port and close the listener and then we don't work even in netstack mode because the OS-level listener is closed
>
> maybe we should only do netstack mode on Android and don't even attempt to do OS-level listener
>
> or really do that just everywhere but iOS

/cc @sailorfrag @JayWStapleton @DentonGentry

**Comment by DentonGentry** [comment]
> or really do that just everywhere but iOS

@twitchyliquid64 is thinking about netstack on iOS now to solve some DNS bugs. The fleet is 94% iOS 15.

**Comment by bradfitz** [comment]
https://github.com/tailscale/tailscale/pull/4465 is submitted and should fix this for 1.24.0. Keeping open until confirmed.
